### DISCUSSION TITLE
Bloc with BottomSheet

### DISCUSSION THREAD

**Discussion by moha-b** [moha-b/LMS#21] [Q&A] [answered]
why we can't use bloc within the bottom sheet?

**Comment by Mahmoud-Abdelaty** [comment, marked as answer]
Actually if you need this bloc only in bottom sheet and nowhere else, the better and cleaner solution is create the StatefullWidget for bottom sheet content, create the Bloc inside this widget in initState() work with bloc in build() method and free resources in dispose() method.
